pkg/plugin/linuxutil/ethtool_stats_linux.go

```diff
@@ -12,11 +12,15 @@ import (
 	"go.uber.org/zap"
 )
 
+const limitValue = 2000
+
 type EthtoolReader struct {
-	l         *log.ZapLogger
-	opts      *EthtoolOpts
-	data      *EthtoolStats
-	ethHandle EthtoolInterface
+	l           *log.ZapLogger
+	opts        *EthtoolOpts
+	data        *EthtoolStats
+	ethHandle   EthtoolInterface
+	unsupported map[string]bool
```

Comment:
`map[string]struct{}` is the right way to use a map as a set. `map[string]bool` incurs overhead to store all those bools. `map[string]struct{}` says "I want to use the map machinery, but I don't actually care about storing anything." You have to use the two-return form of map access to test for the existence of keys like this:

```go
_, ok := er.unsupported[key]
```

And you assign with:

```go
er.unsupported["something"] = struct{}{}
```

Reply:
I would recommend using a proper LRU cache here, in case the interface is deleted, we should be able to reuse those limits, or else at some point we will hit the limit and never process any new interfaces.

Reply:
Yeah @vakalapa is right. LRU is the way to go for this. My comment is more for general Go knowledge in this case.